cmd/main_test.go

```diff
@@ -301,3 +307,75 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
+	})
+
+	Context("When all the namespaces have all the necessary permissions like create, list, watch and delete", func() {
+		BeforeEach(func() {
+			namespaceNames := []string{"test-tenant", "test-tenant-2"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+				createRole(k8sClient, name, "namespace-access", []string{"create", "list", "watch", "delete"})
+				createRoleBinding(k8sClient, "namespace-access-user-binding", name, "user1@konflux.dev", "namespace-access")
+			}			
+			expectedNs = allNamespaces
+		})
+		It("returns all namespaces in the list", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+	})
+
+	Context("When namspace ns3 doesn't have necessary permissions", func() {
+		BeforeEach(func() {
+			var ns3 k8sapi.Namespace
+			ns3, err = createNamespace(k8sClient, "test-tenant-3")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-3", "test-tenant-3", "user3@konflux.dev", "namespace-access-3")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns3: %v", err))
+			allNamespaces = []k8sapi.Namespace{ns3}
+			expectedNs = []k8sapi.Namespace{}
+		})
+		It("doesn't return any namespace", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+	})
+
+	Context("When only namespaces test-tenant and test-tenant-2 has all necessary permissions and other's don't", func() {
+		BeforeEach(func() {
+			namespaceNames := []string{"test-tenant", "test-tenant-2", "test-tenant-3", "test-tenant-4"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}
+			createRole(k8sClient, "test-tenant", "namespace-access", []string{"create", "list", "watch", "delete"})
+			createRole(k8sClient, "test-tenant-2", "namespace-access-2", []string{"create", "list", "watch", "delete"})
+			createRole(k8sClient, "test-tenant-3", "namespace-access-3", []string{"create", "list", "watch"})
+			createRoleBinding(k8sClient, "namespace-access-user-binding", "test-tenant", "user1@konflux.dev", "namespace-access")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-2", "test-tenant-2", "user2@konflux.dev", "namespace-access-2")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-3", "test-tenant-3", "user3@konflux.dev", "namespace-access-3")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-4", "test-tenant-4", "user4@konflux.dev", "namespace-access-4")
```

Comment:
`namespace-access-4` does not exist; by assigned to nothing I meant not having an assignment at all, not having a failed assignment